lib/schemaType.js

```diff
@@ -1356,7 +1356,7 @@ SchemaType.prototype.doValidate = function(value, fn, scope, options) {
     if (err) {
       return;
     }
-    if (ok === undefined || ok) {
+    if (ok === undefined || ok !== false) {
```

Comment:
Maybe instead we add an explicit check for `|| (ok && ok !== '')` ? As written, this change is pretty risky and causes a lot of test failures, partially due to breaking several of our built-in validators. For example, [string `match` validator relies on returning `null` meaning "ok"](https://github.com/Automattic/mongoose/blob/f1ed13f5b4bacbc739ae4c52a19dfcd1e355555a/lib/schema/string.js#L542-L544).

In general, I agree that treating `return ''` as an error but not `return 'test'` is not ideal. But I think this change as written is a bit too risky.